### PR TITLE
Fix toolchain registration call for ScalaTest

### PR DESCRIPTION
**Pull request by liucijus**
Call correct function to register toolchain

### CODE REVIEW

**Comment by ittaiz** [comment]
How come it passed until now?

**Comment by liucijus** [comment]
> How come it passed until now?

In `rules_scala` we use another toolchain, which has configurations for all testing frameworks: https://github.com/bazelbuild/rules_scala/blob/master/WORKSPACE#L50

**Comment by ittaiz** [comment]
Do you think it's a good idea to keep it like this? Feels like a recipe for future bugs

**Comment by liucijus** [comment]
@ittaiz I agree it's an opportunity for bugs. Added an example workspace which serves as an integration test for ScalaTest repository/toolchain helpers.

File: WORKSPACE [hunk 1]
```diff
@@ -119,8 +128,11 @@ format_repositories()
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "45409e6c4f748baa9e05f8f6ab6efaa05739aa064e3ab94e5a1a09849c51806a",
-    url = "https://github.com/bazelbuild/rules_go/releases/download/0.18.7/rules_go-0.18.7.tar.gz",
```
Comment (ittaiz) [inline WORKSPACE:123]: Why did you upgrade rules_go? Because of skylib?

File: testing/scalatest.bzl [hunk 2]
```diff
@@ -3,4 +3,4 @@ def scalatest_repositories():
     pass
 
 def scalatest_toolchain():
-    native.register_toolchain("//testing:scalatest_toolchain")
+    native.register_toolchains("@io_bazel_rules_scala//testing:scalatest_toolchain")
```
Comment (ittaiz) [inline testing/scalatest.bzl:6]: Small question- your previous CI (IIRC) was without `@io_bazel_rules_scala` right? Why the change? What pushed it?

**Review by ittaiz — COMMENTED** [review]
Thanks for the test workspace! Looks really good :) left few questions

File: WORKSPACE (hunk 1 above)
Comment (liucijus) [reply WORKSPACE:123]: yes

File: testing/scalatest.bzl (hunk 2 above)
Comment (liucijus) [reply testing/scalatest.bzl:6]: It was a mistake - all toolchains have to be registered with external name otherwise they may be treated as different

Comment (ittaiz) [reply testing/scalatest.bzl:6]: So it's precaution rather than something actually failed? (Ok with either, trying to understand which)

Comment (liucijus) [reply testing/scalatest.bzl:6]: I tried with the example repo - it doesn't even find the toolchain:
```
ERROR: Analysis of target '//example:example' failed; build aborted: invalid registered toolchain '//testing:scalatest_toolchain': no such package 'testing': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.

```

Comment (ittaiz) [reply testing/scalatest.bzl:6]: Amazing. Great that we have this test workspace

**Pull request merged**